cvmfs/statistics.cc

```diff
@@ -90,19 +90,18 @@ string Statistics::PrintList(const PrintOptions print_options) {
   }
   return result;
 }
+
 string Statistics::PrintJSON() {
-  string result = "{";
   MutexLockGuard lock_guard(lock_);
+
+  JsonStringGenerator json_generator;
   for (map<string, CounterInfo *>::const_iterator i = counters_.begin(),
-       iEnd = counters_.end(); i != iEnd; ++i)
-  {
-    result +=
-      std::string("\"") + i->first + "\":\"" +
-      i->second->counter.ToString() + "\",";
+                                                  iEnd = counters_.end();
+       i != iEnd; ++i) {
+    json_generator.PushBack(i->first, i->second->counter.ToString());
```

Comment:
You need to PushBack with quoted = false, since they need to be interpreted as numbers by the gateway.

Reply:
Thanks!